### PR TITLE
Finall Pack + CI + solved shellcheck errors

### LINKED ISSUE #1104: feat: --create-config option not yet implemented
Currently just a stub marked TODO.
This is just a reminder to either add it or comment out the option (and regen docs so it is not mentioned everywhere) before release

### PR DESCRIPTION
**Pull request by zen0bit**
# New UI finalization

1. Added quickget tests to CI
2. Test made parallel
3. Added SHELL check
4. Added YAML check
5. Fixed introduced regression
6. Finalized arguments and help message
7. Solved shellcheck errors (SC2317 disabled globally)

tests:
1. List all supported OS with releases and editions
8. List infos about all supported OS
9. List all URLs of supported ISOs
10. Test all supported ISOs availability
[Test quickget 🧪](https://github.com/oSoWoSo/quickemu/actions/workflows/test-quickget.yml)
Tests also save results to artifacts for later use (txt in zip)
see it in checks
PS: With parallelization we are under 10 minutes (from 30+)

# Failed:
FAIL: arcolinux-v24.02.01-large           https://mirror.accum.se/mirror/arcolinux.info/iso/v24.02.01/arcolinuxl-v24.02.01-x86_64.iso
FAIL: arcolinux-v24.02.01-small           https://mirror.accum.se/mirror/arcolinux.info/iso/v24.02.01/arcolinuxs-v24.02.01-x86_64.iso
FAIL: arcolinux-v24.05.11-large           https://mirror.accum.se/mirror/arcolinux.info/iso/v24.05.11/arcolinuxl-v24.05.11-x86_64.iso
FAIL: arcolinux-v24.05.11-small           https://mirror.accum.se/mirror/arcolinux.info/iso/v24.05.11/arcolinuxs-v24.05.11-x86_64.iso
FAIL: blendos-v3-gnome                    https://mirror.ico277.xyz/blendos/gnome/blendos-v3-stable-gnome.iso
FAIL: fedora-39-Onyx
FAIL: fedora-39-Sericea
FAIL: rebornos-latest

## Closes #1104 #1150 #1151 #1158

# Help message:
help message

### CODE REVIEW

File: .github/workflows/quickget-tests.yml
Comment (philclifford) [inline .github/workflows/quickget-tests.yml:71]:
```suggestion
            echo -e "\n\n      ${os}" && ./quickget -12345 "${os}" | tee -a osinfo.txt
```

currently we only see zorin in the artifact so you need to append to get all

File: .github/workflows/quickget-tests.yml
Comment (philclifford) [inline .github/workflows/quickget-tests.yml:94]:
```suggestion
            echo -e "\n\n      ${os}" && ./quickget -ua "${os}" | tee -a isoslist.txt
```

File: .github/workflows/quickget-tests.yml
Comment (philclifford) [inline .github/workflows/quickget-tests.yml:118]:
```suggestion
            echo -e "\n\n      ${os}" && ./quickget -ca "${os}" | tee -a isostest.txt
```

**Review by philclifford — CHANGES_REQUESTED** [review]
Need to append the `tee`s where doing "for each whatever" otherwise the assets only tell us about zorin

**Comment by philclifford** [comment]
Something is broken, at least with check. The [CI run](https://github.com/quickemu-project/quickemu/actions/runs/8812654392/job/24188764463?pr=1107#step:3:554) shows a fail for latest minimal but currently we are fine.
 
```
  ./quickget -c openindiana 20231027 minimal
Testing if OI-hipster-minimal-20231027.iso is available... PASS
```

CI is missing the EDITION .  This only happens for OpenIndiana and Siduction (search log for '--' )

Also there's something odd with the reporting of ubuntus - e.g. [line 330](https://github.com/quickemu-project/quickemu/actions/runs/8812654392/job/24188764463?pr=1107#step:3:330) ... "FAIL - ... PASS"

**Comment by zen0bit** [comment]
> Something is broken, at least with check. The [CI run](https://github.com/quickemu-project/quickemu/actions/runs/8812654392/job/24188764463?pr=1107#step:3:554) shows a fail for latest minimal but currently we are fine.
> 
> ```
>   ./quickget -c openindiana 20231027 minimal
> Testing if OI-hipster-minimal-20231027.iso is available... PASS
> ```
> 
> CI is missing the EDITION . This only happens for OpenIndiana and Siduction (search log for '--' )
> 
> Also there's something odd with the reporting of ubuntus - e.g. [line 330](https://github.com/quickemu-project/quickemu/actions/runs/8812654392/job/24188764463?pr=1107#step:3:330) ... "FAIL - ... PASS"

Investigating...
PS: That's exactly why I am pushing CI😉

**Comment by zen0bit** [comment]
Tests will be reworked to work with...

**Comment by zen0bit** [comment]
Maybe we can leave -ca and -ua arguments?

-u and -c will work like:
without argument = will show all OS like -ua -ca doing now
with OS argument = will show all url/checks for an OS
with OS release and/or edition = will show url/check for concrete release and/or edition

Hmm? :thinking:

File: .github/pull_request_template
Comment (flexiondotorg) [inline .github/pull_request_template:1]: Please can you submit this template via its own pull request.

**Review by flexiondotorg — CHANGES_REQUESTED** [review]

**Comment by zen0bit** [comment]
@flexiondotorg rebased on master and all requested solved...

File: .github/workflows/quickget-tests.yml
Comment (zen0bit) [reply .github/workflows/quickget-tests.yml:113]: I see, but we lost nice sorting and separated names in tests then
I like it like that...

I thinking more about..
My previous question:
Maybe we can leave -ca and -ua arguments?

File: .github/workflows/quickget-tests.yml
Comment (zen0bit) [reply .github/workflows/quickget-tests.yml:91]: I see, but we lost nice sorting and separated names in tests then
I like it like that...

I thinking more about..
My previous question:
Maybe we can leave -ca and -ua arguments?

File: .github/workflows/quickget-tests.yml
Comment (zen0bit) [reply .github/workflows/quickget-tests.yml:113]: PS: I wanna also separate checks to individual OS in PR
later...
So green mark will be in PR for (changed OS only) or All checked OS
Depends on change in PR
But that PR template first... #1127

**Comment by zen0bit** [comment]
> Maybe we can leave -ca and -ua arguments?
> Hmm? 🤔

Done

**Comment by zen0bit** [comment]
OK I am done here for now?
Few things still needs to be fixed,
 but that will be just finalization and catching few remaining bugs, I think...

**Comment by zen0bit** [comment]
🚀

**Comment by TuxVinyards** [comment]
Two pp's in SKIPPED.   Otherwise reads as 'Skype'. 

I'll be pleased when all the branches finally get merged. It's been a very active last couple of weeks.

**Comment by zen0bit** [comment]
> Two pp's in SKIPPED. Otherwise reads as 'Skype'.
> 
> I'll be pleased when all the branches finally get merged. It's been a very active last couple of weeks.

I am trying to fix strange counting in checks right after SKIPPED...

**Comment by zen0bit** [comment]
So beacuse @flexiondotorg not merged this yetwhen worked excelently.
And leaved it here for to long..
 I broke it yesterday. Don't know where problem is. Can't figure out why test not working anymore 🤯

**Review by flexiondotorg — CHANGES_REQUESTED** [review]
This pull request is large, sprawling and difficult to review. I kindly request that you refrain from submitting pull requests like this.

In the future, please submit pull requests that are contained to a single fix, a single feature or a focused refactor only. 

Some of the changes in this pull request introduce bugs, some are refactors for the sake of change, some incorrectly change code indentation, and yet there are also some legitimate fixes.

I'll have to think about what to do with this and figure out how to cherry pick the good work that exists in here.

**Comment by zen0bit** [comment]
> This pull request is large, sprawling and difficult to review. I kindly request that you refrain from submitting pull requests like this.
> 
> In the future, please submit pull requests that are contained to a single fix, a single feature or a focused refactor only.
> 
> Some of the changes in this pull request introduce bugs, some are refactors for the sake of change, some incorrectly change code indentation, and yet there are also some legitimate fixes.
> 
> I'll have to think about what to do with this and figure out how to cherry pick the good work that exists in here.

I would like to, but I am almost one year in advance due to your "presence" as maintainer 🤗

What can I do?

Wait another year before all chqnges I want will be in? (And rebase evrything meanwhile thousand times like last half year?)

Sorry but that's maybe bit more your fault "as maintainer" then my.
That I have "A lot to merge"

**Comment by zen0bit** [comment]
> I'll have to think about what to do with this and figure out how to cherry pick the good work that exists in here.

Just point which commits you don't like. I will delete them without effort.. @flexiondotorg

**Comment by zen0bit** [comment]
see...

**Comment by zen0bit** [comment]
Once more, why you don't want merge this? Speak about commit you don't like not a whole.

Everything is working better then in master...

**Comment by flexiondotorg** [comment]
I have a full-time job and a family. Pack it in with the churlish remarks!

**Comment by zen0bit** [comment]
> I have a full-time job and a family. Pack it in with the churlish remarks!

Exactly! Then I wonder even more...
Why are you redoing what's I already done?

Like sorry, but you are here the one, who are wasting our time which could belongs to families instead...

**Comment by flexiondotorg** [comment]
I'm closing this pull request because:
- commits from this branch are already in master
- commits in the branch revert intended functionality in master
- or other pull requests exist for features in the pull request

The only exception is the parallel testing changes, for which I welcome a pull request so it can be reviewed and tested in isolation.

**Pull request closed without merging**
